examples/round_trip.rs

```diff
@@ -59,8 +59,6 @@ async fn main() -> Result<(), pulsar::Error> {
                     data: "data".to_string(),
                 })
                 .await
-                .unwrap()
```

Comment:
It looks like we are breaking the existing public API. isn't it?

Reply:
It is. But for a good intention. In version 5.1.1, it [returns](https://github.com/streamnative/pulsar-rs/blob/v5.1.1/src/producer.rs#L552) a future of receipt when the receipt is already received from the server. It causes misunderstanding of that the client are still waiting for receipt while the client already waited for it. Even the [doc](https://github.com/streamnative/pulsar-rs/blob/v5.1.1/src/producer.rs#L318) was misleading too.

I made it to not return the future, but just the receipt, so it's clear that the receipt is awaited.

Reply:
Hi @cirias 

I am sorry. We should do some nasty things to avoid breaking the public API. Consider that users already have code written using the old approach, But the new version changes it directly. It's kinda pain for existing users.

We can keep the old approach, mark it deprecated, and remove it in the future. Also, we can introduce a brand new method to build an elegant API. 

I would like to know your thoughts. Thanks! :)

Reply:
Make sense. I updated the methods to be backward compatible, and marked them deprecated.